File: apps/python/event_sourcing/command_handler.py
"""Command handler base class."""

from abc import ABC, abstractmethod
from typing import List

from .command import COMMANDS
from .event import EVENTS


class CommandHandler(ABC):
    """Base class for all command handlers."""

    @abstractmethod
    def handle(self, command: COMMANDS) -> List[EVENTS]:
        """Handle a command and return emitted events."""
        pass

File: apps/rust/src/command_handler.rs
//! Command handler trait definition.

use crate::command::Command;
use crate::event::Event;

/// Base trait for all command handlers.
pub trait CommandHandler<C: Command, E: Event> {
    /// Handle a command and return emitted events.
    fn handle(&mut self, command: &C) -> Vec<E>;
}
